jps/model-api/src/org/jetbrains/jps/model/java/JavaModuleSourceRootTypes.java

```diff
@@ -31,6 +31,6 @@
   public static final Set<? extends JpsModuleSourceRootType<?>> TESTS = ContainerUtilRt.newHashSet(JavaSourceRootType.TEST_SOURCE, JavaResourceRootType.TEST_RESOURCE);
 
   public static boolean isTestSourceOrResource(@Nullable JpsModuleSourceRootType<?> type) {
-    return JavaSourceRootType.TEST_SOURCE.equals(type) || JavaResourceRootType.TEST_RESOURCE.equals(type);
+    return type.isForTests();
```

Comment:
This method is located in JavaModuleSourceRootTypes class so users won't expect that it'll return `true` for non-Java test source roots as well. It'll be better to use `isForTests` directly in the places where this method is currently called.

Reply:
Also this code may fail with NullPointerException, and IDEA should show you a warning about that.